src/org/exist/indexing/StreamListener.java

```diff
@@ -43,6 +43,13 @@
         STORE,
 
         /**
+         * Mode for replacing the nodes of a document with another document
+         * this is really a group mode, which is later followed by {@link #REMOVE_ALL_NODES}
+         * and then {@link #STORE}
+         */
+        REPLACE_DOCUMENT,
```

Comment:
it's very confusing, there should be `REPLACE_DOCUMENT` mode and `startIndexDocument` + `endIndexDocument` or just methods `startReplaceDocument` + `endReplaceDocument`. But I can't understand why you adding anything of this if there are `REMOVE_ALL_NODES` or `REMOVE_SOME_NODES` + `STORE` and methods `startIndexDocument` + `endIndexDocument`.